integration-tests/test_selinux.py

```diff
@@ -0,0 +1,287 @@
+        return _check_denials_with_ausearch(self.start_time, self.end_time)
+
+    def get_process_contexts(self):
+        # Get process contexts from execve events during the context manager period.
+        return _check_process_contexts_from_audit(self.start_time, self.end_time)
+
+
+@pytest.mark.tier2
+def test_register_unconfined_t_no_context_change(insights_client):
+    """
+    :id: 91c9c37f-b954-4fc4-84da-a4a2a9dbee9d
+    :title: Test insights-client --register runs in unconfined_t without context change
+    :description:
+        This test verifies that when running insights-client --register as root
+        in unconfined_t SELinux context with SELinux enforcing mode, the insights-core
+        code executes in unconfined_t context without any SELinux context change
+        and without any SELinux denials.
+    :tags: Tier 1
+    :reference: https://issues.redhat.com/browse/CCT-1717
+    :steps:
+        1. Register system using subscription-manager (already done by fixture)
+        2. Set SELinux to enforcing mode (using context manager)
+        3. Verify current context is unconfined_t
+        4. Run insights-client --register as root in unconfined_t context
+        5. Verify registration succeeded
+        6. Check audit logs for process execution contexts (SYSCALL execve events)
+        7. Check audit logs for SELinux AVC denials
+    :expectedresults:
+        1. System is registered (handled by register_subman fixture)
+        2. SELinux is set to enforcing mode and will be restored automatically
+        3. Current context is confirmed as unconfined_t
+        4. insights-client --register completes successfully
+        5. Registration status confirms system is registered
+        6. All insights-client and insights-core processes executed in
+           unconfined_t context
+        7. No SELinux AVC denials found in audit logs
+    """
+    if _get_selinux_mode() == "Disabled":
+        pytest.skip(reason="SELinux is disabled on this system")
+
+    with _selinux_mode("Enforcing"):
+        context = _get_current_context()
+        assert "unconfined_t" in context, f"Expected unconfined_t, got: {context}"
+
+        with SELinuxDenialsChecker() as checker:
+            result = insights_client.run("--register", selinux_context=context)
+            assert result.returncode == 0, f"Registration failed: {result.returncode}"
+
+            # Verify registration
+            def check_registered():
+                status = insights_client.run(
+                    "--status", check=False, selinux_context=context
+                )
+                return status.returncode == 0 and any(
+                    i in status.stdout
+                    for i in ["This host is registered", "Registered"]
+                )
+
+            assert loop_until(check_registered)
+
+        # Verify process contexts from audit logs
+        for proc_name, scontext, running_context in checker.get_process_contexts():
```

Comment:
The logic below looks quite complicated and it's hard to assess whether the right thing is checked. In general, I don't have problem with the `any`, but my head started to spin when I started to think about this code :sweat_smile: Right now I however don't have suggestion for better way other than yet another abstraction layer on top of the audit log. Having that said, let's stick with this implementation if it works and I may introduce changes later if I have some better idea that proves to be working. What do you think?